File: frontend/src/types/models/course.ts
import type { Note } from "./note";
import type { Summarization } from "./summarization";
import type { UploadedFile } from "./uploaded-file";

export type Course = {
  id: string;
  name: string;
  user_id: string;
  description: string;
  created_at: string;
  updated_at: string;
  notes: Note[];
  uploaded_files: UploadedFile[];
  summaries: Summarization[];
};

File: backend/app/database/models/course.py
import uuid
from sqlalchemy import UUID, Column, DateTime, ForeignKey, String, func, null
from sqlalchemy.orm import relationship
from app.database import Base


class Course(Base):
    __tablename__ = "courses"

    id = Column(UUID(as_uuid=True), primary_key=True, index=True, server_default=func.gen_random_uuid(), unique=True, nullable=False)
    name = Column(String, nullable=False)
    description = Column(String)

    user_id = Column(UUID, ForeignKey("users.id"))
    user = relationship("User", back_populates="courses")

    notes = relationship("Note", back_populates="course", cascade="all, delete")

    uploaded_files = relationship("UploadedFile", back_populates="course", cascade="all, delete")

    created_at = Column(
        DateTime(timezone=True), server_default=func.now(), nullable=False
    )
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)
